src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -107,6 +113,8 @@ public override void ReadWrite(BitcoinStream stream)
 
             stream.ReadWrite(ref this.coinstake);
             this.CoinstakeSize = stream.ProcessedBytes - Size - this.MerkleProofSize - this.SignatureSize;
+
+            stream.ReadWrite(ref this.stakeModifierV2);
```

Comment:
I think we don't need this finally, we will not serialize stake modifier just set it.

Reply:
you need it to validate incoming headers: how can you know that when you start a node otherwise?

Reply:
Yes but we will calculate the stake modifier when validating